### PR TITLE
Us107 enter token

### PR DESCRIPTION
**Pull request by Phemus**
This pull request implements the deed token into the borrower personal details table to serve as a lookup. a new endpoint has been added called validate borrower. this is used to validate the the borrower  token provided matches a valid borrower and if true return the deed token associated with that borrower

### CODE REVIEW

File: application/deed/views.py [hunk 1]
```diff
@@ -81,6 +81,13 @@ def create():
             deed.deed = json_doc
 
             deed.save()
+
+            for currentBorrower in deed.deed['borrowers']:
+                borrowerId = currentBorrower['id']
```
Comment (endamccormack) [inline application/deed/views.py]:
```
for currentBorrower in deed.deed['borrowers']:
    borrower = Borrower.get_by_id(currentBorrower['id'])
    borrower.deed_token = deed.token
    borrower.save()
```

would potentially be better, bit of a wasteful use of a variable.

File: unit_tests/test_app.py [hunk 2]
```diff
@@ -30,7 +31,7 @@ def test_model(self):
 
     @mock.patch('application.borrower.model.Borrower.save')
     @mock.patch('application.deed.model.Deed.save')
-    def test_create(self, mock_Borrower, mock_Deed):
+    def test_create_with_invalid(self, mock_Borrower, mock_Deed):
```
Comment (danielmarker) [inline unit_tests/test_app.py:34]: Thanks for applying this fix also - found it in my work also.

**Comment by tsangsimon** [comment]
The saving of the deed.token on the borrower model could be handled better by assigning it when you first create the borrower, especially as this is generated up front. This would save calls to the DB too. lines 84 - 89 are then now unnecessary on the views.py

File: application/borrower/model.py [hunk 3]
```diff
@@ -7,6 +7,7 @@ class Borrower(db.Model):
 
     id = db.Column(db.Integer, primary_key=True)
     token = db.Column(db.String, nullable=False)
+    deed_token = db.Column(db.String, nullable=True)
```
Comment (endamccormack) [inline application/borrower/model.py]: We would prefer a non nullable as this is meant to be like a mandatory foreign key relationship.

**Comment by endamccormack** [comment]
:+1:

**Comment by Kirosoft** [comment]
:+1:

**Pull request merged**